src/shared/keyring-util.c

```diff
@@ -5,34 +5,39 @@
 #include "missing_syscall.h"
 
 int keyring_read(key_serial_t serial, void **ret, size_t *ret_size) {
-        size_t m = 100;
+        long bufsize;
+
+        /* Get the size of the key by passing in a NULL buffer */
+        bufsize = keyctl(KEYCTL_READ, (unsigned long) serial, (unsigned long) NULL, 0, 0);
```

Comment:
hmm, i'd drop this initial call, and just start with a fixed value as before, as this means we can avoid one syscall typically.